File: code/code-interface/src/main/java/cn/moonnow/code/query/VirtualColumnsQuery.java
package cn.moonnow.code.query;

import java.io.Serializable;
import java.util.List;

import lombok.Data;

/**
 * 虚拟列 PRO_VIRTUAL_COLUMNS 查询类
 */
@Data
public class VirtualColumnsQuery implements Serializable {

  private static final long serialVersionUID = 1L;

  protected String virtualColumnsId; // 虚拟列编号

  protected String sourceDtId; // 源表数据库表编号

  protected String targetDtId; // 目标表数据库表编号

  protected String sourceColumnsId; // 源表列编号

  protected String targetColumnsId; // 目标表列编号

  protected String targetDisplayColumnsId; // 目标表显示字段列编号

  protected String displayColumnsAlias; // 显示字段别名

  protected String virtualColumnsSql; // 虚拟列sql

  protected Integer weightOrder; // 排序权重

  protected List<String> virtualColumnsIdAndin; // 虚拟列编号Andin查询

  protected String sourceDtIdAndKeyLike; // 源表数据库表编号AndKeyLike查询

  protected String targetDtIdOrKeyLike; // 目标表数据库表编号OrKeyLike查询

  protected String sourceColumnsIdOrKeyLike; // 源表列编号OrKeyLike查询

  protected String targetColumnsIdOrKeyLike; // 目标表列编号OrKeyLike查询

  protected String targetDisplayColumnsIdOrKeyLike; // 目标表显示字段列编号OrKeyLike查询

  protected String displayColumnsAliasOrKeyLike; // 显示字段别名OrKeyLike查询

  protected String virtualColumnsSqlOrKeyLike; // 虚拟列sqlOrKeyLike查询

}

File: vue-dev-tool/src/view/cn/moonnow/code/virtualColumns/VirtualColumnsQuery.js
export default class VirtualColumnsQuery {
  constructor () {
    this.virtualColumnsId = null // 虚拟列编号
    this.sourceDtId = null // 源表数据库表编号
    this.targetDtId = null // 目标表数据库表编号
    this.sourceColumnsId = null // 源表列编号
    this.targetColumnsId = null // 目标表列编号
    this.targetDisplayColumnsId = null // 目标表显示字段列编号
    this.displayColumnsAlias = null // 显示字段别名
    this.virtualColumnsSql = null // 虚拟列sql
    this.weightOrder = null // 排序权重
    this.virtualColumnsIdAndin = null // 虚拟列编号Andin查询
    this.sourceDtIdAndKeyLike = null // 源表数据库表编号AndKeyLike查询
    this.targetDtIdOrKeyLike = null // 目标表数据库表编号OrKeyLike查询
    this.sourceColumnsIdOrKeyLike = null // 源表列编号OrKeyLike查询
    this.targetColumnsIdOrKeyLike = null // 目标表列编号OrKeyLike查询
    this.targetDisplayColumnsIdOrKeyLike = null // 目标表显示字段列编号OrKeyLike查询
    this.displayColumnsAliasOrKeyLike = null // 显示字段别名OrKeyLike查询
    this.virtualColumnsSqlOrKeyLike = null // 虚拟列sqlOrKeyLike查询
  }
}
